packages/opentelemetry-exporter-jaeger/src/jaeger.ts

```diff
@@ -119,8 +119,9 @@ export class JaegerExporter implements SpanExporter {
     for (const span of thriftSpan) {
       try {
         await this._append(span);
-      } catch (error) {
+      } catch (unknownError) {
         // TODO right now we break out on first error, is that desirable?
+        const error = unknownError instanceof Error ? unknownError : new Error(String(unknownError));
```

Comment:
this might have quite unexpected behaviour including extra failure during error conversion, I don't think you should convert the error at all

Reply:
What cases would `Error(String(unknown))` fail in? AFAIK you can always use `String`.

I was actually thinking the opposite that we should do this everywhere because if something is thrown that isn't an Error it won't be logged which can make debugging really hard.

Reply:
The callstack captured by this `Error` would be quite missleading as it points to the `catch` instead to the `throw` location.

Reply:
The callback (public interface) takes an Error. Changing it to just be `unknown` would fix it, but it would be a breaking change. The callstack may be misleading, but only in the case where something was thrown that was not an `Error`

Reply:
You could try `e.stack = ''` to clear the wrong stack.

Reply:
But then there is no stack at all. Is that better? If a string is thrown and we wrap it in an error, at least the throw location has a decent chance of being in the stack.

Reply:
If a `string` is thrown you catch a string - nothign more. if you wrap it in an `Error` the stack points to the catch location which is for sure not the place where the problem happened. The correct stack is unknown and to my knowlege not recoverable. Therefore it's better to never throw something else then an `Error` and to reject Promises with `Error` instances.
I think no stack is better then a wrong stack.

Reply:
Newest version uses no stack